src/Visualization/Python/PlotDatFile.py

```diff
@@ -0,0 +1,296 @@
+            "one for each function.",
+            choices=['solid', 'dashed', 'dashdot', 'dotted'],
+            default=['solid'])
+        parser.add_argument('--legend-ncols',
+                            required=False,
+                            type=int,
+                            default=1,
+                            help="Number of columns for the legend")
+        parser.add_argument('--x-axis',
+                            required=False,
+                            help="If specified, make the range be the "
+                            "values is this field of the subfile. Defaults to "
+                            "the first column in the dat file.")
+        parser.add_argument('--y-logscale',
+                            required=False,
+                            action='store_true',
+                            help="If specified, set the y-axis to log scale.")
+
+        parser.add_argument('--title',
+                            required=False,
+                            help="Title on the graph.")
+        parser.add_argument('--fontsize',
+                            required=False,
+                            type=int,
+                            default=16,
+                            help="The font size on the plots")
+
+        parser.add_argument(
+            '--labels',
+            required=False,
+            type=json.loads,
+            help="R|A string containing a JSON dictionary to map some of\n"
+            "the functions to specified labels. An example of what\n"
+            "to pass to the argument (including the single quotes)\n"
+            "is:\n"
+            "'{\"Error(DivergenceCleaningField)\":\"$L_2(\\\\Phi)$\",\n"
+            "\"Error(RestMassDensity)\":\"$L_2(\\\\mathcal{E}(\\\\rho))$\"}'")
+
+    args = parser.parse_args()
+
+    # Do not do additional argument parsing if we are only printing file
+    # contents or writing to a text file.
+    if args.legend_only or args.write_dat:
+        return args
+
+    if args.functions is None:
+        args.legend_only = True
+        return args
+
+    num_functions = len(args.functions)
+    num_linestyles = len(args.linestyles)
+
+    # Make sure for each specified label a function with that name exists
+    if args.labels:
+        functions = args.functions
+        for key in args.labels:
+            if not key in functions:
+                parser.error(
+                    "Unknown function '{}' specified with label '{}'. Known"
+                    " functions are:\n{}".format(key, args.labels[key],
+                                                 functions))
+
+    # Make sure the correct number of line styles were supplied
+    if num_linestyles != num_functions and num_linestyles != 1:
+        parser.error(
+            "You must supply either zero, one, or one line style per "
+            "function but parsed {} line styles with {} functions.\n".format(
+                num_linestyles, num_functions))
```

Comment:
I supplied zero arguments to the `--linestyles` option and got an error. I think this should be reworded to "either one linestyle or one linestyle per function".

Also the second half of the sentence doesn't really make sense with the first. How about

```python
parser.error(
            "You must supply either one line style or one line style per "
            "function. However, {} line styles with {} functions were parsed.\n".format(
                num_linestyles, num_functions))
```